docs/releasehistory.rst

```diff
@@ -7,6 +7,60 @@ Releases follow the ``major.minor.micro`` scheme recommended by `PEP440 <https:/
 * ``minor`` increments add features but do not break API compatibility
 * ``micro`` increments represent bugfix releases or improvements in documentation
 
+
+
+0.4.0 - SMIRNOFF Spec Update and Performance Optimization
+---------------------------------------------------------
+
+This update introduces the 0.3 SMIRNOFF spec.
+This spec upgrade is the result of discussions about how to handle the evolution of data and parameter types as further functional forms are added to the SMIRNOFF spec.
+
+The toolkit now contains functionality to "upgrade" the spec of a data source.
+These methods are called automatically when loading an 0.1 or 0.2-spec data source.
+This functionality allows the toolkit to continue to read 0.2-spec files, and also implements backwards-compatibility for 0.1-spec files.
+
+
+WARNING: The 0.1 SMIRNOFF spec did not contain fields for several energy-determining parameters that are exposed in the 0.2 spec.
+Thus, the 0.1-to-0.2 spec conversion process makes assumptions about the values that should be added for the newly-required fields.
+These assumptions are intended to reproduce the spec update process for the smirnoff99Frosst family of force fields.
+If you use the spec conversion functionality for other force field families, please carefully review the warning messages printed during spec conversion to ensure they are providing your desired behavior.
+
+
+This update also greatly improves performance when running ``create_openmm_system`` on large topologies.
+
+SMIRNOFF Spec Changes
+"""""""""""""""""""""
+* All individual sections are now versioned.
+  The top-level ``SMIRNOFF`` tag, containing information like ``aromaticity_model``, ``Author``, and ``Date``, still has a version (currently 0.3).
+  But, to allow for independent development of individual parameter types, each section (`such as ``Bonds``, ``Angles``, etc) has its own version as well (currently all 0.3).
+* All units are now stored in expressions with their corresponding values. For example, distances are now stored as ``1.526*angstrom``, instead of storing the unit separately.
+* The ``potential`` field for ``ProperTorsions`` and ``ImproperTorsions`` tags is no longer ``charmm``, but is rather ``k*(1+cos(periodicity*theta-phase))``.
+  It was pointed out to us that CHARMM-style torsions deviate from this formula when the periodicity of a torsion term is 0.
+
+New features
+""""""""""""
+Adds the following new functions:
+
+* `PR #311 <https://github.com/openforcefield/openforcefield/pull/311>`_:
+    * ``utils/utils.py``: Adds ``convert_0_2_smirnoff_to_0_3``, which takes a 0.2-spec SMIRNOFF data dict, and upgrades it to 0.3.
```

Comment:
Why describe what file it's in? Nobody will use it that way. They'll either use the entry point script (if you provide one) or the Python import path. Explain which are available.

Also, use consistent naming: "0.2-spec SMIRNOFF" should be "SMIRNOFF 0.2". 

Also, the "data dict" structures are very dangerous. We DON'T want people using these. I'd add warnings that these are not guaranteed to be stable forever.

Reply:
Same goes for all the features listed below.

Reply:
Good call. Removed the file paths.

Changed references to SMIRNOFF spec per your suggestion.

Added:
* NOTE: The format of the "SMIRNOFF data dict" above is likely to change significantly in the future. Users that require a stable serialized ForceField object should use the output of ``ForceField.to_string('XML')`` instead.

Changed the to "PR #311: Several new experimental functions." to make it clear that these aren't stable.